pkg/datapath/loader/netlink.go

```diff
@@ -65,11 +65,18 @@ func replaceQdisc(ifName string) error {
 	return nil
 }
 
-// replaceDatapath the qdisc and BPF program for a endpoint
-func (l *Loader) replaceDatapath(ctx context.Context, ifName, objPath, progSec, progDirection string) error {
-	err := replaceQdisc(ifName)
-	if err != nil {
-		return fmt.Errorf("Failed to replace Qdisc for %s: %s", ifName, err)
+// replaceDatapath the qdisc and BPF program for a endpoint or XDP program.
+func replaceDatapath(ctx context.Context, ifName, objPath, progSec, progDirection string, xdp bool, xdpMode string) error {
```

Comment:
Nit: maybe swap the ordering of `xdp bool` and `xdpMode string` so that `xdpMode` can be grouped with the other string args

```suggestion
 func replaceDatapath(ctx context.Context, ifName, objPath, progSec, progDirection, xdpMode string, xdp bool) error {
```

Reply:
I'm not a big fan of grouping args by their type instead of their meaning. Also, having many args under the same type next to each other is more error prone, as a developer might mix them without noticing (vs compiler error).